File: vol1/ch7/LeetCode783.py
class Solution:
    def minDiffInBST(self, root: Optional[TreeNode]) -> int:
        self.ans=100005
        self.pre=None
        self.inorder(root)
        return self.ans

    def inorder(self,root):
        if root:
            self.inorder(root.left)
            if self.pre!=None:self.ans=min(self.ans,abs(root.val-self.pre.val))
            self.pre=root
            self.inorder(root.right)

File: vol1/ch7/LeetCode783.cpp
class Solution {
    int ans;
    TreeNode*pre;
public:
    int minDiffInBST(TreeNode* root) {
        ans=100005;
        pre=NULL;
        inorder(root);
        return ans;
    }

    void inorder(TreeNode*root) {
        if(root!=NULL) {
            inorder(root->left);
            if(pre!=NULL) ans=min(ans,abs(root->val-pre->val));
            pre=root;
            inorder(root->right);
        }
    }
};

